community/kernel/src/main/java/org/neo4j/kernel/impl/newapi/PropertyCursor.java

```diff
@@ -132,32 +134,49 @@ public boolean next()
             }
         }
 
-        while ( block < getNumberOfBlocks() )
+        while ( true )
         {
-            if ( block == -1 )
+            //Figure out number of blocks of record
+            int numberOfBlocks = getNumberOfBlocks();
+            while ( block < numberOfBlocks )
             {
-                block = 0;
+                //We have just read a record, so we are at the beginning
+                if ( block == INITIAL_POSITION )
+                {
+                    block = 0;
+                }
+                else
+                {
+                    //Figure out the type and how many blocks that are used
+                    long current = currentBlock();
+                    PropertyType type = PropertyType.getPropertyTypeOrNull( current );
+                    if ( type == null )
+                    {
+                        break;
+                    }
+                    block += type.calculateNumberOfBlocksUsed( current );
+                }
+                //nothing left, need to read a new record
+                if ( block >= numberOfBlocks || type() == null )
```

Comment:
Use `type` instead of `type()` here as well

Reply:
This is slightly confusing but we cannot reuse type here since after we have done `block +=...` we might be at another property with a different type.